master/buildbot/util/service.py

```diff
@@ -195,23 +197,22 @@ class AsyncMultiService(AsyncService, service.MultiService):
 
     def startService(self):
         service.Service.startService(self)
-        l = []
-        for svc in self:
-            # handle any deferreds, passing up errors and success
-            l.append(defer.maybeDeferred(svc.startService))
-        return defer.gatherResults(l, consumeErrors=True)
+
+        return defer.gatherResults(
+            [defer.maybeDeferred(svc.startService) for svc in self],
+            consumeErrors=True)
 
     def stopService(self):
         service.Service.stopService(self)
-        l = []
-        services = list(self)
-        services.reverse()
-        for svc in services:
-            l.append(defer.maybeDeferred(svc.stopService))
+
         # unlike MultiService, consume errors in each individual deferred, and
         # pass the first error in a child service up to our caller
-        return defer.gatherResults(l, consumeErrors=True)
+        return defer.gatherResults(
+            [defer.maybeDeferred(svc.startService) for svc in reversed(list(self))],
+            consumeErrors=True)
 
+    # XXX(sa2ajj): This method seems to be a full duplicate of the one from
+    # service.MultiService w/ the only difference of returning a deferred.
```

Comment:
Correct -- but more importantly, it returns the deferred from `startService`, which the parent method discards.  So just wrapping the parent and returning `defer.succeed(None)` wouldn't be sufficient.